### PR TITLE
Feature/ads101x

### PR DESCRIPTION
**Pull request by JKazem**
The example uses the same pins as on our custom board on which the driver has been tested. We are using the ads1014 so we have not been able to test the multiplexer functionality.

### CODE REVIEW

**Comment by chris-durand** [comment]
Just for your information, the failure of the Windows CI job is not due to your changes. It is not working right now.

File: src/modm/driver/adc/ads101x.hpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_ADS101X_HPP
+#define MODM_ADS101X_HPP
+
+#include <modm/architecture/interface/i2c_device.hpp>
+
+namespace modm
+{
+    /// @ingroup modm_driver_ads101x
```
Comment (salkinium) [inline src/modm/driver/adc/ads101x.hpp]: We don't indent the namespace to save a bit of horizontal space.

File: src/modm/driver/adc/ads101x.hpp [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+            }
+
+            inline float
+            getVoltage()
+            {
+                return getValue() * lsbSize;
+            }
+
+            uint8_t data[2];
+            float lsbSize;
```
Comment (salkinium) [inline src/modm/driver/adc/ads101x.hpp]: I would instead add a `uint8_t size` that is the index of the lsbSize array.
It adds a single indirection, but on Cortex-M it's practically the same speed.

**Review by salkinium — CHANGES_REQUESTED** [review]
Very nice!
(You need to sync the docs to make the CI happy)

**Review by salkinium — APPROVED** [review]

File: examples/nucleo_g474re/ads101x/main.cpp [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+main()
+{
+    Board::initialize();
+
+    Scl::configure(modm::platform::Gpio::InputType::PullUp);
+    Sda::configure(modm::platform::Gpio::InputType::PullUp);
+    Scl::configure(modm::platform::Gpio::OutputType::OpenDrain);
+    Sda::configure(modm::platform::Gpio::OutputType::OpenDrain);
+
+    I2cMaster::connect<Scl::Scl, Sda::Sda>(I2cMaster::PullUps::Internal);
```
Comment (chris-durand) [inline examples/nucleo_g474re/ads101x/main.cpp]: The gpio modes are also configured inside `connect()`.

```suggestion
    I2cMaster::connect<Scl::Scl, Sda::Sda>(I2cMaster::PullUps::Internal);
```

**Comment by salkinium** [comment]
Do you want to squash your fix into 723ce50 and rebase on top of latest develop?

**Review by chris-durand — APPROVED** [review]

**Comment by salkinium** [comment]
Btw, thanks for contributing back, I see a lot of cool stuff in the DanSTAR repo, so I'm happy to see it appear here too ;-P

**Pull request merged**